saleor/graphql/product/mutations/products.py

```diff
@@ -539,6 +744,18 @@ def clean_sku(cls, product_type, cleaned_input):
             elif models.ProductVariant.objects.filter(sku=input_sku).exists():
                 raise ValidationError({"sku": "Product with this SKU already exists."})
 
+    @classmethod
+    def get_instance(cls, info, __qs: QuerySet = None, **data):
+        """Prefetch related fields that are needed to process the mutation."""
+        # If we are updating an instance and want to update its attributes,
+        # prefetch them.
+        if data.get("id") and data.get("attributes"):
```

Comment:
I am not sure that we really want to complicate whole mutation logic only to pass `qs` to the lowest layer of the `ModelMutation`. I think that maintaining it and understanding will cost us a lot.
Product with Attributes is a special case so maybe we should focus on modifying `get_instance` on the Product level without the need to applying changes to core Mutations also.
Maybe something similar to this?
```
    @classmethod
    def get_instance(cls, info, **data):
        """Prefetch related fields that are needed to process the mutation."""
        # If we are updating an instance and want to update its attributes,
        # prefetch them.
        if data.get("id") and data.get("attributes"):
            qs = Product.objects.prefetch_related(
                "product_type__product_attributes",
                "attributes__assignment__attribute__values",
            )
            object = qs.object(pk=data.get("id")).first()
            if object:
                return object
        return super().get_instance(info, **data)
```
@maarcingebala What do you think?

Reply:
Agreed.